### ISSUE TITLE
Push vs pull UX

### ISSUE THREAD

**Issue by ackernaut** [ParabolInc/parabol#3391]
**TL;DR** - In response to Jordan’s “_Oh_ 💩, _I can’t forget to talk about this_!” shower idea. Let’s imagine Parabol as a push-pull system (edit: I admit the analogy uses loose semantics):

Our multi-meeting milestone has largely been about pulling folks into a meeting. We want the primary action to be starting or joining a meeting. This is great for the short term as this is the primary value of our system today.

In the future, we could implement the “push” side of our system. One idea is how to afford a solid UX and UI for getting thoughts, ideas, tensions, etc. easily into the system. I’ll call it notes for now to have a semantic.

I take notes all day. Some of those represent different types of needs: something to do, something to remember or reference, etc.

Drunken house drawing: In Parabol everything starts as a simple note (card, comment, etc). I can push it into a future check-in as an agenda item. I can push it to a running or future retro. I can push it to my next 1:1. I can push it to my task board, etc. Or it can just hang out in my notes to be searched later.

Now our system is more balanced. When I join my next retro I could find a few notes I added a week ago and pull them in if I didn’t quite make the push at the time. When I have an idea that I was jamming on I call pull on folks to make an async decision etc.

Basically Parabol is a place to take structured notes where we can upgrade the bandwidth in structured ways. And one can easily go both ways from personal roles to teams to organization and back with these push/pull loops and references.

![New Doc 2019-11-25 12 31 00](https://user-images.githubusercontent.com/307286/69665307-bdbd9300-104f-11ea-8f90-ab27896d9d16.jpg)

**Push**: write down most things, aka the easiest first step, and structure the notes as needed
**Pull**: others into structured activities to review things as needed aka upgrading the bandwidth

Parabol = structured notebook/knowledge + structured collaboration (with emergent structured organizations)

**Comment by ackernaut** [comment]
A [user](https://app.hubspot.com/sales/3888472/contact/5142551) 🔒  wrote about their pre-existing proccess in Trello:
>my number one feature request would be to have an 'icebox' or ability to add items to a retro board before the meeting. We currently have a slack integration with trello that we use to flag issues as they arise so we remember to cover them in the retro, so then have to look at the trello board in the retro and copy items across to parabol, having something that's open from the point the last retro ends to let people add items ahead of time would be amazing. 

And explained more:

>Generally we just have an 'ice box', just a single column that things go into, good or bad and we organise them into the appropriate columns when the retro format is decided, potential solution could be the ability to add/flag things before a retro then at the stage of the retro where you add your tickets you are first prompted to categorise, update or ignore anything you've added since the last retro, it'd also be nice if the team, or someone was shown the tickets raised by anyone not attending the retro. The reason we do it is that are often things that we want to flag in a retro, but by the time it comes round we've forgotten things. 

>Currently we used the trello slack integration so it's super easy in our team channel to just say `/trello add This thing was annoying` at the time then we can be reminded of it during retro. 

I explained that they could open a retro in advance. They said they’d try that. They commented on missing out on the social check-in which perhaps they like to do. (see the email thread on Hubspot).

My own comment is that this represents a user’s need to push something easily and quickly without having a space for it to land yet, or without having to think too hard about where it’s going. Later that can be pulled in easily. An idea: during the retro phase they can quickly search through their private notes in the input.

**Comment by ackernaut** [comment]
### Capturing and processing in the push/pull analogy

What types of things do we capture? The drunken note, to bring out the knives (disclaimer, I time-boxed shaping some recurring thinking into something cohesive):

(See Figma sketches: 1) [capturing and pushing notes](https://www.figma.com/file/tucu7HKkbkog701BCd4nnK?node-id=41%3A1501&utm_campaign=team&utm_source=slack&utm_medium=message&utm_content=file_comment#16466700) 2) [pulling notes into meetings](https://www.figma.com/file/tucu7HKkbkog701BCd4nnK?node-id=41%3A1501&utm_campaign=team&utm_source=slack&utm_medium=message&utm_content=file_comment#16466801))

Jot something down? A tension? A fleeting note? A reminder to do something?
- Do I tag it, or push it, as a thought or topic to bring up in a 1:1 or retro?
- Do I push it to an agenda in order to find next steps, resulting in a task?
- Do I mean to shape something into a proposal or give background that needs a decision?

An over-simplified frame: what we capture leans more toward knowing or doing. What we process may result in e.g. explicitly writing down a team process vs. identifying next-step tasks when we’re stuck on something. However, a decision, for example, feels like it falls somewhere up the middle—where knowing and doing intersect. We know and assume some things and we need to decide where the razor falls which informs our actions (e.g. the process for handling this case in the future, the acceptance criteria for implementing a feature, the design of a system or machine, etc.)

Some frameworks to illustrate knowing vs doing:

GTD vs Zettelkästen
https://gettingthingsdone.com/what-is-gtd/
https://clerestory.netlify.com/zk/

### Principles
Some concepts at play: 1) recall vs recognition 2) links and nodes 3) push/pull

We capture nodes in writing. We tag or push to appropriate spaces to make links. This forms a map, a history of activity. The nodes don’t have to live in any single space, but they can be referenced in any given context. [capturing and pushing notes](https://www.figma.com/file/tucu7HKkbkog701BCd4nnK?node-id=41%3A1501&utm_campaign=team&utm_source=slack&utm_medium=message&utm_content=file_comment#16466700)

We begin to pull on others through processes to review the nodes. We upgrade the bandwidth as needed, from async to sync, from text to video, from a simple topic with a thread to a structured format that helps us find useful prompts for framing and useful cycles of divergence and convergence for generating and improving our ideas. (One caveat, I think that pulling on others within the system will encourage usage of the system. I think there’s a win when others start building the muscle of pushing their own nodes into the system because of its usefulness. Killer features delight along the way.)

We can look to templates, or our own notes (esp. those marked as being something to bring up in a retro or 1:1) or even bring in integrated objects like issues from GitHub, or a card from Guru, as we build an agenda for discussion in the process of moving forward (next steps, decision, capture knowledge, adopt a process). [pulling notes into meetings](https://www.figma.com/file/tucu7HKkbkog701BCd4nnK?node-id=41%3A1501&utm_campaign=team&utm_source=slack&utm_medium=message&utm_content=file_comment#16466801))

### The system vs the UX
The object system is atomic, flexible, defaults to open, and favors composability. On the other hand, any given loop UX is opinionated and solves for specific problems. It uses objects from the system in constrained ways. The optionality of the object system can be exposed via settings (think default no. of votes), or the abstraction can be hidden away (e.g. the system allows for any depth of replies but in this UX we only allow 1 level deep). Loops are probably not always linear and clean, hence relying on the principle of links and nodes for organic organization. Rather, the goal should be making progress toward a completed swing. The system automatically builds the links and history for when you want to trace steps or generate and visualize data insights (trends, etc.)

**Comment by jordanh** [comment]
Scrubbed, iceboxing
